tls/s2n_connection.c

```diff
@@ -513,6 +503,11 @@ int s2n_connection_free_handshake(struct s2n_connection *conn)
     return 0;
 }
 
+/* An idempotent operation which initializes values on the connection.
+ *
+ * An existing connection should be reset only after all I/O is completed and `s2n_shutdown`
+ * has been called.
```

Comment:
By this, are you trying to warn users not to wipe a connection while they're still using it? I'm not sure that needs to be said, and if it does need to be said it should be in the usage guide, not here.

Maybe it would be more helpful to explain how this method is called and the expected effect on the connection? You want to provide guidance to developers on where to add their init code. Like, "Called in order to reuse a connection structure for a new connection. Should wipe any persistent memory, free any temporary memory, and set all fields back to their defaults."